tests/pds/dataservice_test.go

```diff
@@ -272,29 +181,154 @@ var _ = Describe("{UpgradeDataServiceVersion}", func() {
+				logrus.Infof("storage replicas used %v ", config.Spec.StorageOptions.Replicas)
+				logrus.Infof("cpu requests used %v ", config.Spec.Resources.Requests.CPU)
+				logrus.Infof("memory requests used %v ", config.Spec.Resources.Requests.Memory)
+				logrus.Infof("storage requests used %v ", config.Spec.Resources.Requests.Storage)
+				logrus.Infof("No of nodes requested %v ", config.Spec.Nodes)
+				logrus.Infof("volume group %v ", storageOp.VolumeGroup)
+				logrus.Infof("version/images used %v ", config.Spec.Version)
+
+				Expect(resourceTemp.Resources.Requests.CPU).Should(Equal(config.Spec.Resources.Requests.CPU))
+				Expect(resourceTemp.Resources.Requests.Memory).Should(Equal(config.Spec.Resources.Requests.Memory))
+				Expect(resourceTemp.Resources.Requests.Storage).Should(Equal(config.Spec.Resources.Requests.Storage))
+				Expect(resourceTemp.Resources.Limits.CPU).Should(Equal(config.Spec.Resources.Limits.CPU))
+				Expect(resourceTemp.Resources.Limits.Memory).Should(Equal(config.Spec.Resources.Limits.Memory))
+				repl, err := strconv.Atoi(config.Spec.StorageOptions.Replicas)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(storageOp.Replicas).Should(Equal(int32(repl)))
+				Expect(storageOp.Filesystem).Should(Equal(config.Spec.StorageOptions.Filesystem))
+				Expect(config.Spec.Nodes).Should(Equal(replicas))
+				for version, build := range dataServiceVersionBuildMap {
+					Expect(config.Spec.Version).Should(Equal(version + "-" + build[index]))
 				}
 			}
-		}()
+		}
+	})
+
+	defer func() {
+		Step("Delete the worload generating deployments", func() {
+			if DataService == "Cassandra" || DataService == "PostgreSQL" {
+				err = pdslib.DeleteK8sDeployments(dep.Name, namespace)
+			} else {
+				err = pdslib.DeleteK8sPods(pod.Name, namespace)
+			}
+			Expect(err).NotTo(HaveOccurred())
+
+		})
+		Step("Delete created deployments")
+		for _, dep := range deployments {
+			for index := range dep {
+				resp, err := pdslib.DeleteDeployment(dep[index].GetId())
+				Expect(err).NotTo(HaveOccurred())
+				Expect(resp.StatusCode).Should(BeEquivalentTo(http.StatusAccepted))
+			}
+		}
+	}()
+}
+
+var _ = Describe("{UpgradeDataServiceVersion}", func() {
+
+	JustBeforeEach(func() {
+		if !DeployAllDataService {
+			supportedDataServices = append(supportedDataServices, DataService)
+			for _, ds := range supportedDataServices {
+				logrus.Infof("supported dataservices %v", ds)
+			}
+			Step("Get the resource and app config template for supported dataservice", func() {
+				dataServiceDefaultResourceTemplateIDMap, dataServiceNameIDMap, err = pdslib.GetResourceTemplate(tenantID, supportedDataServices)
+				Expect(err).NotTo(HaveOccurred())
+
+				dataServiceNameDefaultAppConfigMap, err = pdslib.GetAppConfTemplate(tenantID, dataServiceNameIDMap)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(dataServiceNameDefaultAppConfigMap).NotTo(BeEmpty())
+			})
+
+		} else {
```

Comment:
Is the validation required here? if it else condition then value must be true only